--- src/interfaces/jdbc/org/postgresql/jdbc1/AbstractJdbc1DatabaseMetaData.java.orig
+++ src/interfaces/jdbc/org/postgresql/jdbc1/AbstractJdbc1DatabaseMetaData.java
@@ -3620,8 +3620,11 @@
 			" NULL AS FILTER_CONDITION "+
 			from+
 			" WHERE ct.oid=i.indrelid AND ci.oid=i.indexrelid AND a.attrelid=ci.oid AND ci.relam=am.oid "+
-			where+
-			" AND ct.relname = '"+escapeQuotes(tableName)+"' ";
+			where;
+
+		if (tableName != null && ! "".equals(tableName)) {
+			sql += " AND ct.relname = '"+escapeQuotes(tableName)+"' ";
+		}
 
 		if (unique) {
 			sql += " AND i.indisunique ";
