--- org/postgresql/util/PSQLException.java.in~	2005-11-24 03:29:23.000000000 +0100
+++ org/postgresql/util/PSQLException.java.in	2007-01-06 15:16:39.000000000 +0100
@@ -22,7 +22,8 @@
     public PSQLException(String msg, PSQLState state, Throwable cause)
     {
         super(addCauseToMessage(msg, cause), state == null ? null : state.getState());
-        initCause(cause); // Calls either the 1.4 Throwable impl or our impl below.
+        @NOINITCAUSE@ initCauseA(cause); // Calls either the 1.4 Throwable impl or our impl below.
+        @INITCAUSE@ initCause(cause);
 
     }
 
@@ -70,7 +71,7 @@
     }
 
     // Dummy initCause() for pre-1.4 JVMs
-    @NOINITCAUSE@ public void initCause(Throwable cause) {
+    @NOINITCAUSE@ public void initCauseA(Throwable cause) {
         @NOINITCAUSE@ }
 
 
